streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -841,12 +838,37 @@ public void apply(final StreamTask task) {
         if (e != null) {
             throw e;
         }
+    }
 
+    private void commitAllStandbyTasks() {
+        RuntimeException e = null;
         for (final StandbyTask task : standbyTasks.values()) {
-            commitOne(task);
+            try {
+                commitOne(task);
+            } catch (final RuntimeException t) {
```

Comment:
Just a `KafkaException`, i'll convert to that. 
It's difficult to recycle the `performOnStreamTask` without refactoring a whole bunch of other things. Not sure it's needed. `performOnStreamTask` is called several times for active tasks. For standby tasks we just need to commit them.